File: Programmers/Collection/2019 Kakao Developer Winter Internship/크레인인형뽑기게임.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    stack<int> s;
    int answer = 0;
    
    for (int i = 0; i < moves.size(); i++) {
        for (int j = 0; j < board.size(); j++) {
            if (board[j][moves[i] - 1] == 0) continue;
            else {
                if (!s.empty() && (s.top() == board[j][moves[i] - 1])) {
                    s.pop();
                    answer += 2;
                }
                else {
                    s.push(board[j][moves[i] - 1]);
                }
                board[j][moves[i] - 1] = 0;
                break;
            }
        }
    }
    return answer;
}

File: Programmers/난이도모음/LEVEL 1/크레인인형뽑기게임.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> bucket = new Stack<>();
        
        for (int move : moves) {
            for (int i = 0; i < board.length; i++) {
                if (board[i][move-1] == 0) continue;
                else {
                    int doll = board[i][move-1];
                    if (!bucket.empty() && bucket.peek() == doll) {
                        answer += 2;
                        bucket.pop();
                    }
                    else {
                        bucket.push(doll);
                    }
                    board[i][move-1] = 0;
                    break;
                }
            }
        }
        return answer;
    }
}